src/modules/products/products.service.ts

```diff
@@ -231,4 +291,31 @@ export class ProductsService {
 
     return mappedProducts;
   }
+
+  async deleteProduct(vendorId: string, productId: string): Promise<void> {
+    try {
+      const product = await this.productRepository.findOne({
+        where: { id: productId, vendorId },
+      });
+
+      let deleteResponse;
+
+      if (product.status === Status.INACTIVE) {
+        deleteResponse = await this.productRepository.delete(productId);
+      } else if (product.status === Status.PUBLISHED) {
+        deleteResponse = await this.productRepository.softDelete(productId);
+        await this.cartRepository.delete({ productId });
+        await this.wishlistRepository.delete({ productId });
+      }
+
+      if (!deleteResponse || !deleteResponse.affected) {
+        throw new NotFoundException(Errors.PRODUCT_NOT_FOUND);
+      }
+    } catch (error) {
+      if (error instanceof NotFoundException) {
+        throw error;
+      }
+      throw new InternalServerErrorException(Errors.FAILED_TO_DELETE_PRODUCT);
+    }
+  }
```

Comment:
you need to use transaction

Reply:
added transaction, never used it before, useful thing